src/components/Cart.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react' // eslint-disable-line no-unused-vars
+import { ListGroup, Card, Button } from 'react-bootstrap' // eslint-disable-line no-unused-vars
+import CartItem from './CartItem'
+import { SessionContext } from '../App'
+import StripeCheckout from 'react-stripe-checkout'
+
+function Cart () {
+  const { state, dispatch } = React.useContext(SessionContext)
+
```

Comment:
here better desctructure the state to get the properties that you need:

`const { cartCounter, cartTotal } = state`
 or 
`const { state: { cartCounter }, dispatch } = React.useContext(SessionContext)`

I see that this can be applied to several components

Reply:
Thanks for the observations Cesar!

Reply:
About the context implementation I think that its a good one, and about the testing, You can wrap your test inside the provider and mock the values of your state, something like this:

```JSX
<Context.Provider value={{ useReducer(...) }}>
      ...
 </Context.Provider>
```

This way, you're components have access to your "state" and you cant test you're components ;)

Reply:
Great Cesar, thanks for the advice and clarification :100: